projects/microsoft.com/azure-cli/package.yml

```diff
@@ -0,0 +1,58 @@
+distributable:
+   url: https://github.com/Azure/azure-cli/archive/refs/tags/azure-cli-{{version}}.tar.gz
+   strip-components: 1
+
+# if there’s a github then we can parse the versions
+versions:
+  github: Azure/azure-cli/tags  # reads github tags from github
+  strip: /^azure-cli-/
+
+dependencies:
+  openssl.org: 1.1
+  python.org: ^3.10
+  sourceware.org/libffi: '*'
+
+build:
+  dependencies:
+    tea.xyz/gx/cc: c99
+    tea.xyz/gx/make: '*'
+    rust-lang.org: '*'
+    pip.pypa.io: '*'
+    linux:
+      freedesktop.org/pkg-config: '*'
+
+  script: |
+    python -m venv {{prefix}}/venv
+    cd {{prefix}}/venv
+    pip install {{prefix}}/../src-v{{version}}/src/azure-cli
+    pip install {{prefix}}/../src-v{{version}}/src/azure-cli-core
+    pip install {{prefix}}/../src-v{{version}}/src/azure-cli-telemetry
+    mkdir -p ../bin
+
+    cat <<EOF > ../bin/"az"
+      #!/bin/sh
+      export VIRTUAL_ENV="\$(cd "\$(dirname "\$0")"/.. && pwd)/venv"
+      TEA_PYTHON="\$(which python)"
+      TEA_PYHOME="\$(dirname "\$TEA_PYTHON")"
+      cat <<EOSH > \$VIRTUAL_ENV/pyvenv.cfg
+        home = \$TEA_PYHOME
+        include-system-site-packages = false
+        executable = \$TEA_PYTHON
+      EOSH
+      find "\$VIRTUAL_ENV"/bin -maxdepth 1 -type f | xargs \\
+        sed -i.bak "1s|.*|#!\$VIRTUAL_ENV/bin/python|"
+      rm "\$VIRTUAL_ENV"/bin/*.bak
+      ln -sf "\$TEA_PYTHON" "\$VIRTUAL_ENV"/bin/python
+      exec "\$VIRTUAL_ENV"/bin/az "\$@"
+    EOF
+
+    chmod +x ../bin/"az"
```

Comment:
Does our existing `python-venv.sh` script not work for this?

Reply:
no because in the root of the repo there's no setup.py. azure has three subrepos (see line 27-29)

Reply:
It works locally, but I feel like something's not right here

Reply:
Hm. What if you ran them back-to-front:
```sh
for venv in azure-cli{-telemetry,-core,}; do
  python-venv.sh "{{prefix}}"/az
done
```
?

Just thinking that gets the most battle-testing in terms of venv management.

Reply:
Doesn't work like that unfortunately, or I don't understand where you'd like to add it. We need some things that `python-venv.sh` is doing but not all

Reply:
Roger. I figure that'd build all three parts into the venv, and overwrite the final linkage each time; but if it's doesn't work, it doesn't work.

Reply:
Ok I may have found a way: 

```
for venv in azure-cli{-telemetry,-core,}; do
  SRCROOT={{prefix}}/../src-v{{version}}/src/"$venv" python-venv.sh "{{prefix}}"/az
done
```

Reply:
Magical. I love standardization.